docs/docs/troubleshooting.md

```diff
@@ -1,5 +1,175 @@
+
+### Upgrade the instance size the `general` node group
+
+The `general` node group, or node pool, is the (usually only one) node that hosts most of the pods that Nebari relies on for its core services: `hub`, `conda-store`, `proxy` and so on. We have attempted to "min-max" the instance size of the node: large enough so that the initial deployment will work out of the box, while keeping total cloud compute costs to a minimum.
+
+Although each cloud provider has different names and hourly prices for their compute nodes, the default `general` node group in `qhub-config.yaml` has 2 vCPU and 8 GB of memory.
+
+> Given the possible destructive nature of resizing this node group, we **highly recommend** backing up your cluster before attempting.
+
+Based on some testing, clusters running on Google Kubernetes Engine (GKE) appear to be amenable to in-place upgrades of the `general` node instance size. Unfortunately, this does not seem to be the case with the other cloud providers, and attempting to do so for AWS and Azure will likely result in a catastrophic destruction of your cluster.
+
+| Cloud Provider | `general` node upgrade possible? |
+| :------------- | :------------------------------- |
+| AWS            | No (Danger!)                     |
+| Azure          | No (Danger!)                     |
+| Digital Ocean  | No                               |
+| GCP            | Yes                              |
+
+If modifying the resource allocation for the `general` node in-place is absolutely necessary, try increasing the maximum number of nodes for the `general` node group. This will mean two nodes (reserved for the `general` node group) will always be running, ultimately increasing the operating cost of the cluster.
+
+Alternatively, you can backup your cluster, destroy it, specify the new instance size in your `qhub-config.yaml`, and redeploy.
+
+### Use a DNS provider other than CloudFlare
+
+CloudFlare is one of the most commonly used DNS providers for Nebari, so to some it may seem as if it is the *only* DNS provider Nebari supports. This is NOT the case! Please check out our "How to" section, paying particular attention to the domain registry section.
+
+### Add system packages to a user's (jupyterlab) image
+
+In some cases, you may wish to customize the default user's JupyterLab image, including installing some system packages via apt (or other OS package manager) or adding some JupyterLab extensions.
+
+Nebari uses its own registered docker images for `jupyterhub`, `dask`, and `jupyterlab` services by default, but this can be changed by:
+
+ 1. Building your own docker images, and
+ 2. Including the DockerHub register hash into `qhub-config.yaml`.
+
+### Provide individual users with unique environments and cloud instance types
```

Comment:
What you have here is basically describing nebari in general. I think what you mean is something like:

>Nebari allows for admins to set up user groups which each have access to specific environments and server instance types. This allows for a fine-tuned management of your cloud resources. For example, you can create a special user group for a team of ML engineers which provides access to GPUs and a pytorch environment. This will prevent inexperienced users from accidentally consuming expensive resources. 

and so on...

Reply:
True, but this came out of a Discussion where that seemed to be non-obvious https://github.com/Quansight/qhub/discussions/675 .

Reply:
He's running into some sort of error, unrelated to this concept. I guess we can just leave this in though.